flocker/control/httpapi.py

```diff
@@ -868,7 +868,9 @@ def container_configuration_response(application, node):
     result.update(ApplicationMarshaller(application).convert())
     # Configuration format isn't quite the same as JSON format:
     if u"volume" in result:
-        result[u"volumes"] = [result.pop(u"volume")]
+        volume = result.pop(u"volume")
+        result[u"volumes"] = [{u"dataset_id": volume[u"dataset_id"],
+                               u"mountpoint": volume[u"mountpoint"]}]
```

Comment:
:question: This is done in order to remove the `maximum_size` key from the volume, right? In order to comply with our output schema. Maybe add an explanatory comment.